### DISCUSSION TITLE
Tx submission timing (+during congestion)

### DISCUSSION THREAD

**Discussion by xdzurman** [CardanoSolutions/ogmios#187] [Q & A] [answered]
Hello!

Please, when you submit an `invalid` tx through `txClient.submitTx(tx)` during network congestion, how does the validation work? Do you immediately get the response that the `tx is malformed` / `utxos are spent` / `...` ? Or does this validation occur later, so you have to wait e.g. 5 minutes regardless of tx submission success or failure?

Thanks!

**Comment by KtorZ** [comment, marked as answer]
It should occur immediately since the validation is _local_. A transaction is only added to a node's mempool if and only if it is well-formed and valid. Thus, this happens before any congestion. This is also true for phase-2 validations (Plutus scripts) and scripts evaluation in general.

**Reply by xdzurman** [reply]
Thanks for a blazing fast answer :)
